### DISCUSSION TITLE
assign_skin_depth for Maxwell 2D

### DISCUSSION THREAD

**Discussion by xiaoxiao1987** [ansys/pyaedt#4737] [Q&A] [answered]
I am trying to call mesh.assign_skin_depth for my Maxwell 2D simulation, but I got an error message saying " This method is not supported in the current AEDT design type". I checked the source code and found this method checks if the application is either HFSS or Maxwell 3D, or it throws the error. Maxwell 2D should allow skin depth setting as long as I am assigning to edges instead of surfaces I believe.

**Comment by gmalinve** [comment, marked as answer]
Hi @xiaoxiao1987,

Thanks for opening this discussion.
I just open an issue for this matter: #4802 . 
We could enhance the already existing method _assign_skin_depth_ or create a new one to allow the assignment of skin depth layered based mesh to edges.
